lib/mongo/cluster.rb

```diff
@@ -213,5 +256,17 @@ def direct_connection?(address)
     def addition_allowed?(address)
       !@topology.single? || direct_connection?(address)
     end
+
+    def servers_list
+      @servers_update.synchronize do
+        @servers
+      end
```

Comment:
Does this return a copy of the list? Doesn't it need to return a copy in order to avoid overlapping with a thread that's updating the servers while holding the servers_update lock? E.g. it seems like if one thread is in scan! and slowly iterating the list while another thread is in remove_hosts, removing servers while holding the lock, you can still get a runtime err.